nil pointer dereference when trying to purge pages deployment

### Confirmation

- [x] My issue isn't already found on the issue tracker.
- [x] I have replicated my issue using the latest version of cloudflare-utils and it is still present.

### cloudflare-utils version

1.4.3

### Go environment
```
go env
AR='ar'
CC='gcc'
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_ENABLED='1'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
CXX='g++'
GCCGO='gccgo'
GO111MODULE=''
GOAMD64='v1'
GOARCH='amd64'
GOAUTH='netrc'
GOBIN=''
GOCACHE='/home/weston/.cache/go-build'
GOCACHEPROG=''
GODEBUG=''
GOENV='/home/weston/.config/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFIPS140='off'
GOFLAGS=''
GOGCCFLAGS='-fPIC -m64 -pthread -Wl,--no-gc-sections -fmessage-length=0 -ffile-prefix-map=/tmp/go-build3482973190=/tmp/go-build -gno-record-gcc-switches'
GOHOSTARCH='amd64'
GOHOSTOS='linux'
GOINSECURE=''
GOMOD='/dev/null'
GOMODCACHE='/home/weston/go/pkg/mod'
GONOPROXY=''
GONOSUMDB=''
GOOS='linux'
GOPATH='/home/weston/go'
GOPRIVATE=''
GOPROXY='https://proxy.golang.org,direct'
GOROOT='/usr/lib/go-1.24'
GOSUMDB='sum.golang.org'
GOTELEMETRY='local'
GOTELEMETRYDIR='/home/weston/.config/go/telemetry'
GOTMPDIR=''
GOTOOLCHAIN='auto'
GOTOOLDIR='/usr/lib/go-1.24/pkg/tool/linux_amd64'
GOVCS=''
GOVERSION='go1.24.2'
GOWORK=''
PKG_CONFIG='pkg-config'
```

### Expected output

The command should purge all of the Cloudflare Pages deployments

### Actual output

The program panics:
```
cloudflare-utils --api-token $API_KEY --account-id $ACCOUNT_ID purge-deployments --delete-project --project $PROJECT_NAME --lots-of-deployments

panic: panic: runtime error: invalid memory address or nil pointer dereference
	stacktrace:
	goroutine 6 [running]:
	runtime/debug.Stack()
		runtime/debug/stack.go:26 +0x5e
	github.com/sourcegraph/conc/panics.NewRecovered(0x1, {0x934d40, 0xe909f0})
		github.com/sourcegraph/conc@v0.3.0/panics/panics.go:59 +0x85
	github.com/sourcegraph/conc/panics.(*Catcher).tryRecover(0xc0003fe430)
		github.com/sourcegraph/conc@v0.3.0/panics/panics.go:28 +0x5f
	panic({0x934d40?, 0xe909f0?})
		runtime/panic.go:792 +0x132
	golang.org/x/time/rate.(*Limiter).wait(0xc0001d2f00, {0x0, 0x0}, 0x1, {0x0?, 0xc00070f040?, 0xea6ee0?}, 0xa175b0)
		golang.org/x/time@v0.9.0/rate/rate.go:264 +0x1f1
	golang.org/x/time/rate.(*Limiter).WaitN(0xc0001d2f00, {0x0, 0x0}, 0x1)
		golang.org/x/time@v0.9.0/rate/rate.go:249 +0x50
	golang.org/x/time/rate.(*Limiter).Wait(...)
		golang.org/x/time@v0.9.0/rate/rate.go:234
	github.com/cloudflare/cloudflare-go.(*API).makeRequestWithAuthTypeAndHeadersComplete(0xc0001b2370, {0x0, 0x0}, {0x9e4a2f, 0x6}, {0xc0002b8000, 0x84}, {0x0, 0x0}, 0x4, ...)
		github.com/cloudflare/cloudflare-go@v0.115.0/cloudflare.go:250 +0x3d8
	github.com/cloudflare/cloudflare-go.(*API).makeRequestWithAuthTypeAndHeaders(0x0?, {0x0?, 0x0?}, {0x9e4a2f?, 0xa?}, {0xc0002b8000?, 0x0?}, {0x0?, 0x0?}, 0x4, ...)
		github.com/cloudflare/cloudflare-go@v0.115.0/cloudflare.go:196 +0x5a
	github.com/cloudflare/cloudflare-go.(*API).makeRequestWithAuthType(...)
		github.com/cloudflare/cloudflare-go@v0.115.0/cloudflare.go:180
	github.com/cloudflare/cloudflare-go.(*API).makeRequestContext(0xc000482080?, {0x0?, 0x0?}, {0x9e4a2f?, 0x3?}, {0xc0002b8000?, 0x0?}, {0x0?, 0x0?})
		github.com/cloudflare/cloudflare-go@v0.115.0/cloudflare.go:172 +0x4e
	github.com/cloudflare/cloudflare-go.(*API).DeletePagesDeployment(0xc0001b2370, {0x0, 0x0}, 0x0?, {{0x7ffccb58ff99, 0xe}, {0xc0000f2023, 0x24}, 0x1})
		github.com/cloudflare/cloudflare-go@v0.115.0/pages_deployment.go:252 +0x19f
	main.RapidPagesDeploymentDelete.func1()
		github.com/Cyb3r-Jak3/cloudflare-utils/cmd/cloudflare-utils/utils.go:144 +0xa9
	main.RapidPagesDeploymentDelete.(*ResultPool[...]).Go.func2()
		github.com/sourcegraph/conc@v0.3.0/pool/result_pool.go:34 +0x2c
	github.com/sourcegraph/conc/pool.(*Pool).worker(0x0?)
		github.com/sourcegraph/conc@v0.3.0/pool/pool.go:154 +0x69
	github.com/sourcegraph/conc/panics.(*Catcher).Try(0x0?, 0xc0004d2fd0?)
		github.com/sourcegraph/conc@v0.3.0/panics/panics.go:23 +0x42
	github.com/sourcegraph/conc.(*WaitGroup).Go.func1()
		github.com/sourcegraph/conc@v0.3.0/waitgroup.go:32 +0x4d
	created by github.com/sourcegraph/conc.(*WaitGroup).Go in goroutine 1
		github.com/sourcegraph/conc@v0.3.0/waitgroup.go:30 +0x73
	
	

goroutine 1 [running]:
github.com/sourcegraph/conc/panics.(*Catcher).Repanic(...)
	github.com/sourcegraph/conc@v0.3.0/panics/panics.go:38
github.com/sourcegraph/conc.(*WaitGroup).Wait(0xc0003fe420)
	github.com/sourcegraph/conc@v0.3.0/waitgroup.go:42 +0x3c
github.com/sourcegraph/conc/pool.(*Pool).Wait(0xc0003fe420)
	github.com/sourcegraph/conc@v0.3.0/pool/pool.go:79 +0x30
github.com/sourcegraph/conc/pool.(*ResultPool[...]).Wait(...)
	github.com/sourcegraph/conc@v0.3.0/pool/result_pool.go:41
main.RapidPagesDeploymentDelete({0xc000258c88, {0x0, 0x0}, 0xc00047e150, {0x7ffccb58ff99, 0xe}, {0xc000237808, 0x4, 0x4}})
	github.com/Cyb3r-Jak3/cloudflare-utils/cmd/cloudflare-utils/utils.go:158 +0x34e
main.PruneDeploymentsRoot({0xab4918, 0xc000252870}, 0xc000258c88)
	github.com/Cyb3r-Jak3/cloudflare-utils/cmd/cloudflare-utils/prune-deployments.go:147 +0x4c5
main.PurgeDeploymentsScreen({0xab4918, 0xc000252870}, 0xc000258c88)
	github.com/Cyb3r-Jak3/cloudflare-utils/cmd/cloudflare-utils/purge-deployments.go:53 +0xb0
github.com/urfave/cli/v3.(*Command).run(0xc000258c88, {0xab4918, 0xc000252870}, {0xc0001d2730, 0x5, 0x5})
	github.com/urfave/cli/v3@v3.3.3/command_run.go:336 +0x261e
github.com/urfave/cli/v3.(*Command).run(0xc000259408, {0xab4918, 0xc0002521e0}, {0xc000190140, 0xa, 0xa})
	github.com/urfave/cli/v3@v3.3.3/command_run.go:274 +0x1eae
github.com/urfave/cli/v3.(*Command).Run(...)
	github.com/urfave/cli/v3@v3.3.3/command_run.go:94
main.main()
	github.com/Cyb3r-Jak3/cloudflare-utils/cmd/cloudflare-utils/main.go:117 +0x3f
```

### Steps to reproduce

1. Attempt to delete a Cloudflare Pages deployment that has 100+ deployments.

### Link to debug output

https://gist.github.com/WestonReed/81742e8d028c02f4c4e941d9545afd3a

--- a/.github/renovate.json
+++ b/.github/renovate.json
@@ -1,10 +1,8 @@
 {
   "$schema": "https://docs.renovatebot.com/renovate-schema.json",
   "extends": [
-    "config:recommended",
-    "helpers:pinGitHubActionDigestsToSemver",
-    "schedule:earlyMondays",
-    "github>Cyb3r-Jak3/renovate-configs//assign-me.json"
+    "github>Cyb3r-Jak3/renovate-configs//recommend-extended.json",
+    "github>Cyb3r-Jak3/renovate-configs//my-registry.json"
   ],
   "postUpdateOptions": ["gomodTidy", "gomodUpdateImportPaths"]
 }

--- a/cmd/cloudflare-utils/dns-cleaner.go
+++ b/cmd/cloudflare-utils/dns-cleaner.go
@@ -192,7 +192,7 @@ func DownloadDNS(ctx context.Context, c *cli.Command) error {
 }
 
 // UploadDNS makes the changes to DNS records based on the dns file.
-func UploadDNS(ctx context.Context, c *cli.Command) error {
+func UploadDNS(_ context.Context, c *cli.Command) error {
 	dnsFilePath := c.String(dnsFileFlag)
 	if !common.FileExists(dnsFilePath) {
 		return fmt.Errorf("no DNS file found at '%s'", dnsFilePath)

--- a/cmd/cloudflare-utils/dns-cleaner.go
+++ b/cmd/cloudflare-utils/dns-cleaner.go
@@ -230,7 +230,7 @@ func UploadDNS(ctx context.Context, c *cli.Command) error {
 		fmt.Printf("Dry Run: Would have removed %d records\n", len(toRemove))
 		return nil
 	}
-	removeErrors := RapidDNSDelete(ctx, zoneResource, toRemove)
+	removeErrors := RapidDNSDelete(zoneResource, toRemove)
 	errorCount = len(removeErrors)
 
 	if errorCount == 0 {

--- a/cmd/cloudflare-utils/dns-purge.go
+++ b/cmd/cloudflare-utils/dns-purge.go
@@ -63,7 +63,7 @@ func DNSPurge(ctx context.Context, c *cli.Command) error {
 		return nil
 	}
 
-	errors := RapidDNSDelete(ctx, zoneResource, records)
+	errors := RapidDNSDelete(zoneResource, records)
 	errorCount := len(errors)
 
 	if errorCount == 0 {

--- a/cmd/cloudflare-utils/prune-deployments.go
+++ b/cmd/cloudflare-utils/prune-deployments.go
@@ -18,7 +18,6 @@ const (
 
 type pruneDeploymentOptions struct {
 	c                   *cli.Command
-	ctx                 context.Context
 	ResourceContainer   *cloudflare.ResourceContainer
 	ProjectName         string
 	SelectedDeployments []cloudflare.PagesProjectDeployment
@@ -74,7 +73,7 @@ func buildPruneDeploymentsCommand() *cli.Command {
 }
 
 // PruneDeploymentsScreen is the entry point for the prune-deployments command.
-// It handles parsing the CLI arguments, and then calls PruneDeploymentsRoot.
+// It handles parsing the CLI arguments and then calls PruneDeploymentsRoot.
 func PruneDeploymentsScreen(ctx context.Context, c *cli.Command) error {
 	logger.Info("Staring prune deployments")
 	if err := CheckAPITokenPermission(ctx, PagesWrite); err != nil {

--- a/cmd/cloudflare-utils/utils.go
+++ b/cmd/cloudflare-utils/utils.go
@@ -109,13 +109,13 @@ func DeploymentsPaginate(params PagesDeploymentPaginationOptions) ([]cloudflare.
 
 // RapidDNSDelete is a helper function to delete DNS records quickly.
 // Uses a pool of goroutines to delete records in parallel.
-func RapidDNSDelete(ctx context.Context, rc *cloudflare.ResourceContainer, dnsRecords []cloudflare.DNSRecord) map[string]error {
+func RapidDNSDelete(rc *cloudflare.ResourceContainer, dnsRecords []cloudflare.DNSRecord) map[string]error {
 	p := pool.NewWithResults[bool]()
 	results := make(map[string]error)
 	p.WithMaxGoroutines(maxGoRoutines)
 	for _, dnsRecord := range dnsRecords {
 		p.Go(func() bool {
-			err := APIClient.DeleteDNSRecord(ctx, rc, dnsRecord.ID)
+			err := APIClient.DeleteDNSRecord(context.Background(), rc, dnsRecord.ID)
 			if err != nil {
 				logger.WithError(err).Warningf("Error deleting DNS record: %s\n", dnsRecord.ID)
 				results[dnsRecord.ID] = err
@@ -141,7 +141,7 @@ func RapidPagesDeploymentDelete(options pruneDeploymentOptions) map[string]error
 	p.WithMaxGoroutines(goRoutines)
 	for _, deployment := range options.SelectedDeployments {
 		p.Go(func() bool {
-			err := APIClient.DeletePagesDeployment(options.ctx, options.ResourceContainer, cloudflare.DeletePagesDeploymentParams{
+			err := APIClient.DeletePagesDeployment(context.Background(), options.ResourceContainer, cloudflare.DeletePagesDeploymentParams{
 				ProjectName:  options.ProjectName,
 				DeploymentID: deployment.ID,
 				Force:        true,

